.github/workflows/__build-workflow.yaml

```diff
@@ -137,9 +137,6 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Configure Git for private repositories (temporary until making KGO public)
-        run: git config --global url."https://${{ secrets.gh-pat }}@github.com".insteadOf "https://github.com"
-
```

Comment:
Do we reuse those pipelines in EE, don't we?
So, it's required for https://github.com/Kong/go-enterprise-license that is private and required by EE build

Reply:
~For KGO EE this would not be necessary because workflows on Github already contain the token that has access to checkout the repo that they are running for (I'm pretty sure that's the case).~

This would be necessary as you've pointed out for the ee license repo though to check it out. Let me revert that change

Reply:
I think that comment should be updated, because 
> temporary until making KGO public

is no longer true

Reply:
4821c04f